graspy/inference/latent_distribution_test.py

```diff
@@ -125,37 +153,75 @@ def __init__(
+            msg = "{} is invalid number of bootstraps, must be positive"
             raise ValueError(msg.format(n_bootstraps))
 
         if not isinstance(workers, int):
             msg = "workers must be an int, not {}".format(type(workers))
             raise TypeError(msg)
-        elif workers <= 0:
-            msg = "{} is invalid number of workers, must be greater than 0"
+        elif workers == 0 or workers < 1:
+            msg = (
+                "{} is invalid number of workers, must be positive, or -1 (to use all)"
+            )
             raise ValueError(msg.format(workers))
 
+        if not isinstance(order_correction, bool):
+            msg = "order_correction must be a bool, not {}".format(
+                type(order_correction)
+            )
+            raise TypeError(msg)
+
         super().__init__(n_components=n_components)
 
         if callable(metric):
             metric_func = metric
         else:
-            if metric == "gaussian":
-                metric_func = _medial_gaussian_kernel
+            if metric in _VALID_DISTANCES:
+                if test == "hsic":
+                    msg = (
+                        f"{test} is a kernel-baed test, but {metric} "
+                        "is a distance. results may not be optimal. it is "
+                        "recomended to use either a different test or one of "
+                        f"the kernels: {_VALID_KERNELS} as a metric."
+                    )
+                    warnings.warn(msg, UserWarning)
+
+                def metric_func(X, Y=None, metric=metric, workers=None):
+                    return pairwise_distances(X, Y, metric=metric, n_jobs=workers)
+
+            elif metric == "gaussian":
+                if test != "hsic":
+                    msg = (
+                        f"{test} is a distance-baed test, but {metric} "
+                        "is a kernel. results may not be optimal. it is "
+                        "recomended to use either a hisc as a test or one of "
+                        f"the distances: {_VALID_DISTANCES} as a metric."
+                    )
+                    warnings.warn(msg, UserWarning)
+                metric_func = gaussian
             else:
+                if test != "hsic":
+                    msg = (
+                        f"{test} is a distance-baed test, but {metric} "
+                        "is a kernel. results may not be optimal. it is "
+                        "recomended to use either a hisc as a test or one of "
+                        f"the distances: {_VALID_DISTANCES} as a metric."
+                    )
+                    warnings.warn(msg, UserWarning)
 
                 def metric_func(X, Y=None, metric=metric, workers=None):
-                    return pairwise_distances(X, Y, metric=metric)
+                    return pairwise_kernels(X, Y, metric=metric, n_jobs=workers)
 
         self.test = KSample(test, compute_distance=metric_func)
```

Comment:
should probably document this (i know was not changed in this PR). Also I guess it's weird that we are overwriting an attribute "test" with an object

Reply:
just to clarify, you are saying to document the .test?

Reply:
well im saying it's odd that normally one would be able to pass in parameters, and then in sklearn world, can always access the values of those parameters by doing object.parameter. In this case that's not true. I won't worry about it for now since we are already talking about what to do with the API

Reply:
ah... yeah, i understood you. yes, a way to resolve this would be to instantiate the hyppo object in fit rather than in init. happy not to worry about it due to the coming api change.